arquillian-webcam-video-recorder/src/main/java/org/arquillian/extension/recorder/video/webcam/impl/WebcamRecorder.java

```diff
@@ -0,0 +1,177 @@
+
+    public static final int DEFAULT_FRAMERATE = 20;
+
+    public static final int MIN_FRAMERATE = 0;
+
+    public static final int MAX_FRAMERATE = 100;
+
+    private int frameRate = DEFAULT_FRAMERATE;
+
+    private final Dimension screenBounds;
+
+    private volatile boolean running = false;
+
+    private Thread thread;
+
+    private File recordedVideo = null;
+
+    private Timer timer;
+
+    private VideoConfiguration configuration;
+
+    public WebcamRecorder(WebcamVideoConfiguration configuration) {
+        Validate.notNull(configuration, "Video configuration is null!");
+        this.configuration = configuration;
+        this.frameRate = configuration.getFrameRate();
+        screenBounds = Toolkit.getDefaultToolkit().getScreenSize();
+    }
+
+    public void startRecording(final File toFile) {
+        if (isRecording()) {
+            throw new IllegalStateException("Already recording");
+        }
+
+        recordedVideo = toFile;
+
+        try {
+            recordedVideo.createNewFile();
+        } catch (IOException ex) {
+            throw new RuntimeException("Unable to create file to which video will be saved: " + recordedVideo.getAbsolutePath());
+        }
+
+        running = true;
+
+        thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+
+                Webcam webcam = Webcam.getDefault();
+                webcam.setViewSize(screenBounds);
+                webcam.open();
+
+                try {
+                    SequenceEncoder enc = new SequenceEncoder(recordedVideo);
+                    timer = new Timer();
+                    timer.schedule(new TestTimeoutTask(), TimeUnit.SECONDS.toMillis(configuration.getTestTimeout()));
+
+                    while (running) {
+                        BufferedImage bgrScreen = webcam.getImage();
+                        enc.encodeImage(bgrScreen);
+
+                        try {
+                            Thread.sleep(500 / frameRate);
+                        } catch (InterruptedException ex) {
+                            logger.log(Level.WARNING, "Exception occured during video recording", ex);
+                        }
+                        if (!running) {
+                            enc.finish();
+                        }
+                    }
+
+                } catch (IOException ex) {
+                    logger.log(Level.WARNING, "Exception occured during video recording", ex);
+                }
+
+            }
+        });
+        thread.start();
+    }
+
+    public synchronized Video stopRecording() {
+        if (!running) {
+            throw new IllegalStateException("It seems you are not recording anything yet. Please call startRecording()");
+        }
+
+        running = false;
+
+        try {
+            thread.join();
+        } catch (InterruptedException ignored) {
+            throw new RuntimeException("Unable to stop video recording.");
```

Comment:
So you prefer I remove the throws and add a logger line? It seems I should do both things to not continue the execution of down lines. Also we will need to change it in Desktop.